src/Orleans/Core/GrainFactory.cs

```diff
@@ -24,30 +24,37 @@ MIT License
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading.Tasks;
 using Orleans.CodeGeneration;
-using Orleans.Core;
 using Orleans.Runtime;
+using GrainInterfaceData = Orleans.CodeGeneration.GrainInterfaceData;
 
 namespace Orleans
 {
-    using System.Threading.Tasks;
-
     /// <summary>
     /// Factory for accessing grains.
     /// </summary>
     public class GrainFactory : IGrainFactory
     {
         /// <summary>
+        /// The runtime code generator.
+        /// </summary>
+        private static readonly Lazy<IRuntimeCodeGenerator> CodeGenerator = new Lazy<IRuntimeCodeGenerator>(LoadCodeGenerator);
```

Comment:
I'd also add that if Orleans was using a DI framework, you wouldn't need the `static Lazy`, you would just inject `IRuntimeCodeGenerator` into `GrainFactory` via the constructor and ask the DI container to provide an instance in a lazy fashion. Hopefully someone will find the time to implement DI in Orleans, the code will be and look much simpler.